ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -129,7 +129,7 @@ void setWhereExpression(final Expression whereExpression) {
     return ImmutableList.copyOf(groupByExpressions);
   }
 
-  void addGroupByExpressions(final Set<Expression> expressions) {
+  void addGroupByExpressions(final Collection<Expression> expressions) {
```

Comment:
shouldn't grouping expressions be a unique set though ? what does it mean to `group by a,a,b,a` ?  it's possible i'm entirely misunderstanding what this piece of code does though, rusty on this stuff ;)

Reply:
That's a good point Nick. The old code didn't actually enforce this as `addGroupByExpressions` was called multiple times.  

I've corrected the code and added tests to cover this.  Now:

```sql
CREATE TABLE OUTPUT AS SELECT COUNT(*) FROM TEST GROUP BY DATA, DATA;
```

Results in an error:

```
Duplicate GROUP BY expression: TEST.DATA
```

Reply:
nice!